### ISSUE TITLE
NativeAOT doesnt produce warnings for adding Requires on a static constructor

### ISSUE THREAD

**Issue by tlakollo** [dotnet/runtime#73172]
### Description

Given that static constructor invocation happens in particular scenarios controlled by the runtime, the Requires attribute is not allowed to be placed in the static constructor. Instead, the recommended way to suppress warnings coming from the static constructor is to annotate the class with Requires.
Tools like linker and analyzer display warnings to the user when the attribute is placed in the static constructor, NativeAOT needs to be able to present those warnings too.

### Reproduction Steps

Add any of the Requires attributes to a static constructor
```C#
public class MyClass{
    // IL3056
    // IL2116
    // IL3004
    [RequiresDynamicCode ("some message")]
    [RequiresUnreferencedCode ("some message")]
    [RequiresAssemblyFiles ("some message")
    public static MyClass {
    }
}

### Expected behavior

The commented warnings are generated

### Actual behavior

No warning is generated

### Regression?

_No response_

### Known Workarounds

_No response_

### Configuration

_No response_

### Other information

_No response_

**Comment by MichalStrehovsky** [comment]
@tlakollo  could you milestone this and remove untriaged? It shows up in queries.

**Comment by vitek-karas** [comment]
This has been fixed in #84089.
